### PR TITLE
Add Systemd config for self-hosted server

### PR DESCRIPTION
**Pull request by svenstaro**
Fixes https://github.com/atuinsh/atuin/issues/1864.

## Checks
- [x] I am happy for maintainers to push small adjustments to this PR, to speed up the review cycle
- [x] I have checked that there are no existing pull requests for the same thing

### CODE REVIEW

**Comment by ellie** [comment]
The env makes sense, but it might be better to just use the config file. We autocreate with the defaults which _hopefully_ leads to less people asking for help with things

Setting ATUIN_CONFIG_DIR=/etc/atuin and ensuring that directory is created would make that work

**Comment by svenstaro** [comment]
I reworked this following the suggestions in #1864. I think this is probably good to go.

File: systemd/atuin.service [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[Unit]
```
Comment (ellie) [inline systemd/atuin.service:1]: Could we also use the name `atuin-server` please? I think it's fairly important to make the distinction, especially considering the Atuin Daemon that will soon also have a systemd unit file

**Comment by ellie** [comment]
Just to clarify here too, please make it very clear in the unit file that this is for running _atuin server_, and not anything client related

File: systemd/atuin.service (hunk 1 above)
Comment (svenstaro) [reply systemd/atuin.service]: Alright, renamed this and the sysusers file.

**Comment by svenstaro** [comment]
@ellie Friendly ping (for this and https://github.com/atuinsh/docs/pull/45) :) Are you cool with this now?

**Comment by ellie** [comment]
Both look good now, thank you!

**Review by ellie — APPROVED** [review]
Thank you!

Seeing as this is your first time contributing, if you would like a holographic contributors-only Atuin sticker, then please fill out this [form!](https://notionforms.io/forms/contributors-stickers)

We do also have a [Discord](https://discord.gg/Fq8bJSKPHh) if you'd like to ask any questions, or just fancy hanging out!

**Pull request merged**